common/src/main/java/com/microsoft/identity/common/internal/ui/webview/certbasedauth/NfcSmartcardCertBasedAuthChallengeHandler.java

```diff
@@ -77,24 +103,27 @@ protected SmartcardPinDialog.PositiveButtonListener getSmartcardPinDialogPositiv
             @Override
             public void onClick(@NonNull final char[] pin) {
                 //For NFC, we need another dialog prompting the user to hold the smartcard to the phone again.
-                mDialogHolder.showSmartcardNfcPromptDialog(new SmartcardNfcPromptDialog.CancelCbaCallback() {
+                mDialogHolder.showSmartcardNfcPromptDialog(new ICancelCbaCallback() {
                     @Override
                     public void onCancel() {
-                        mDialogHolder.dismissDialog();
-                        mTelemetryHelper.setResultFailure(USER_CANCEL_MESSAGE);
-                        request.cancel();
+                        getGeneralCancelCbaCallback(request).onCancel();
+                        mCbaManager.stopDiscovery(mActivity);
                     }
                 });
                 mCbaManager.setConnectionCallback(new IConnectionCallback() {
                     @Override
                     public void onCreateConnection() {
                         mDialogHolder.showSmartcardNfcLoadingDialog();
                         if (mCbaManager.isDeviceChanged()) {
-                            //In a future version, an error dialog with a custom message could be shown here instead of a general error.
-                            indicateGeneralException(methodTag, new Exception("Device connected via NFC is different from initially connected device."));
-                            request.cancel();
                             clearPin(pin);
-                            onPausedSmartcardDiscovery();
+                            request.cancel();
```

Comment:
Does it matter if we clear pin or cancel request first?

Reply:
Doesn't matter. Tried to keep it consistent by clearing pin first then cancelling, but there isn't any meaning behind it.